crates/ef-testing/src/models/case.rs

```diff
@@ -97,35 +97,37 @@ impl BlockchainTestCase {
         tx_signed.signature = signature;
 
         let execution_result = sequencer.execute_transaction(tx_signed);
-        log_execution_result(&execution_result, &self.case_name, &self.case_category);
-
-        let retdata = execution_result
-            .map(extract_execution_retdata)
-            .unwrap_or_default();
-
-        Ok(retdata)
+        let output = extract_output_and_log_execution_result(
+            &execution_result,
+            &self.case_name,
+            &self.case_category,
+        )
+        .unwrap_or_default();
+
+        Ok(output)
     }
 
     fn handle_post_state(
         &self,
         sequencer: &mut KakarotSequencer,
-        retdata: Option<String>,
+        output: EVMOutput,
     ) -> Result<(), RunnerError> {
         let wallet = LocalWallet::from_bytes(&self.secret_key.0)
             .map_err(|err| RunnerError::Other(vec![err.to_string()].into()))?;
         let sender_address = wallet.address().to_fixed_bytes();
 
-        let eth_validation_failed = retdata
-            .map(|retdata| retdata == "Kakarot: eth validation failed")
-            .unwrap_or_default();
+        let maybe_revert_reason = String::from_utf8(output.return_data.as_slice().to_vec());
+        let eth_validation_failed = maybe_revert_reason.map_or(false, |revert_reason| {
+            revert_reason == "Kakarot: eth validation failed"
+        });
 
         // Get gas_used and base_fee from RLP block - as in some cases, the block header is not present in the test data.
         let sealed_block = SealedBlock::decode(&mut self.block.rlp.as_ref())
             .map_err(RunnerError::RlpDecodeError)?;
         let sealed_header = sealed_block.header.unseal();
 
         let coinbase = sealed_header.beneficiary;
-        let gas_used: U256 = U256::from(sealed_header.gas_used);
+        let gas_used: U256 = U256::from(output.gas_used);
```

Comment:
I'm not sure why we do this: I think we should compare the two values, not just replace one with the other

Reply:
Used `expected` and `actual` values, compared them and pushed in `diff` message if not coherent.